projects/vdk-core/src/vdk/internal/builtin_plugins/builtin_hook_impl.py

```diff
@@ -50,24 +50,42 @@ def vdk_initialize(self, context: CoreContext) -> None:
 
         Configuration can override an attempt/op/execution id.
 
-        Otherwise each is auto-generated in format:
-        * Attempt id format is "execution_id-random_suffix"
-        * Execution id format is "op_id-random_suffix"
-        * Op Id is random string.
+        Attempt id and Execution id are extracted from env variables.
+        If not present they are auto generated.
+
+        If auto generated:
+        * Attempt id format is uuid + timestamp + five random chars.
+        * Execution id is attempt id minus last 6 chars.
+        * Op id is equal to execution id.
         """
         op_id = context.configuration.get_value(vdk_config.OP_ID)
         execution_id = context.configuration.get_value(vdk_config.EXECUTION_ID)
         attempt_id = context.configuration.get_value(vdk_config.ATTEMPT_ID)
 
-        if not op_id:
-            op_id = str(int(time.time()))
+        if not attempt_id:
+            # Generate attempt id if execution id not present.
+            if not execution_id:
```

Comment:
I don't think we have such a case. We should focus only on what we have.

Reply:
I did refactored/add a unit test. Could you please check again?